src/npctalk_funcs.cpp

```diff
@@ -213,6 +214,118 @@ void talk_function::insult_combat( npc &p )
     p.set_attitude( NPCATT_KILL );
 }
 
+void talk_function::bionic_install( npc &p )
+{
+    std::vector<item *> bionic_inv = g->u.items_with( []( const item & itm ) {
+        return itm.is_bionic();
+    } );
+    if( bionic_inv.empty() ) {
+        popup( _( "You have no bionics to install!" ) );
+        return;
+    }
+
+    std::vector<itype_id> bionic_types;
+    std::vector<std::string> bionic_names;
+    for( auto &bio : bionic_inv ) {
+        if( std::find( bionic_types.begin(), bionic_types.end(), bio->typeId() ) == bionic_types.end() ) {
+            if( !g->u.has_bionic( bionic_id( bio->typeId() ) ) || bio->typeId() ==  "bio_power_storage" ||
+                bio->typeId() ==  "bio_power_storage_mkII" ) {
+
+                bionic_types.push_back( bio->typeId() );
+                bionic_names.push_back( bio->tname() + " - $" + to_string( bio->price( true ) * 2 / 100 ) );
+            }
+        }
+    }
+    // Choose bionic if applicable
+    int bionic_index = 0;
+    bionic_names.push_back( _( "Cancel" ) );
+    bionic_index = menu_vec( false, _( "Which bionic do you wish to have installed?" ),
+                             bionic_names ) - 1;
+    if( bionic_index == ( int )bionic_names.size() - 1 ) {
```

Comment:
Please use `static_cast<int>` instead of C-style `( int )` cast.

Reply:
Fixed.